src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Components/ScopeStack.cs

```diff
@@ -40,55 +44,27 @@ public void OpenComponentScope(CodeRenderingContext context, string name, string
         {
             context.CodeWriter.Write($"({parameterName}) => ");
         }
-
-        scope.LambdaScope = context.CodeWriter.BuildLambda(BuilderVarName);
+        OpenScope(context);
     }
 
-    public void OpenTemplateScope(CodeRenderingContext context)
-    {
-        var currentScope = new ScopeEntry("__template", ScopeKind.Template);
-        _stack.Push(currentScope);
+    public void OpenTemplateScope(CodeRenderingContext context) => OpenScope(context);
 
-        // Templates always get a lambda scope, because they are defined as a lambda.
-        OffsetBuilderVarNumber(1);
-        currentScope.LambdaScope = context.CodeWriter.BuildLambda(BuilderVarName);
+    private void OpenScope(CodeRenderingContext context)
```

Comment:
Is there any reason to have both of these methods now?

Reply:
Well, `OpenScope` is private. `OpenComponentScope` and `OpenTemplateScope` are two public 
methods, each does a slightly different thing, the shared code is in the private `OpenScope`.
That seems fine to me, or what do you suggest?

Reply:
I don't see why we wouldn't just make `OpenScope` `internal`, and delete `OpenTemplateScope`.

Reply:
One reason could be that `OpenTemplateScope` is more descriptive. So there are currently two kinds of scopes - component and template. The methods as defined now perhaps tell that story more clearly  than what you suggest.

But I don't know, I'm okay doing what you say if @chsienki agrees

Reply:
Yeah, I originally left it that was as it seemed more descriptive, but I'm fine just having `OpenScope` given that the template scope doesn't actually do anything different.

Reply:
@333fred what do you think? Can we leave this as it's more descriptive or you think it's better to have just `OpenScope`?

Reply:
I don't feel too strongly. I'll approve and you can decide.